### PR TITLE
fix(git-import): updates spec for git import

### PR DESCRIPTION
**Pull request by invincibleJai**
- updates spec for git import form
- updates spec for pipelines

### CODE REVIEW

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx [hunk 1]
```diff
@@ -8,76 +8,109 @@ import { ImportFlowForm } from './../ImportFlowForm';
...
+    expect(wrapper.find("AppNameSelector").length).toBe(1);
+    expect(wrapper.find("AppNameSelector").prop("namespace")).toBe("default");
+  });
+
+  it("Button create should be disabled and has bsStyle primary", () => {
+    const wrapper = renderImportForm();
+    expect(wrapper.find("Button").at(0).prop('className')).toBe("disabled");
+    expect(wrapper.find("Button.disabled").prop("bsStyle")).toBe("primary");
+    expect(wrapper.find("Button.disabled").contains('Create')).toBe(true);
+  });
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:100]: Assert that the button is disabled by checking for the html attribute `disabled` if it's not there and it's only a class, then there's a bug!
Do not make assertions on constant styles like `bsStyle=primary`.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx [hunk 2]
```diff
@@ -8,76 +8,109 @@ import { ImportFlowForm } from './../ImportFlowForm';
...
+    const wrapper = renderImportForm();
+    expect(wrapper.length).toBe(1);
+  });
 
+  it("renders Form, FormGroup and buttons", () => {
+    const wrapper = renderImportForm();
+    expect(wrapper.find("Form").length).toBe(1);
+    expect(wrapper.find("FormGroup").length).toBe(4);
+    expect(wrapper.find("Button").length).toBe(2);
+  });
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:89]: This has little value as a unit test.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx [hunk 3]
```diff
@@ -8,76 +8,109 @@ import { ImportFlowForm } from './../ImportFlowForm';
...
+describe('ImportFlowForm:: shallow ', () => {
+  function renderImportForm(args = {}) {
+    const prop = { ...props, ...args };
+    return shallow(<ImportFlowForm {...prop} />);
+  }
+
+  it("renders component", () => {
+    const wrapper = renderImportForm();
+    expect(wrapper.length).toBe(1);
+  });
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx]: This unit test adds no value.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 1 above)
Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:100]: yes `disabled` only in class at least in DOM object captured from shallow.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 3 above)
Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx]: will remove as if other test's are passing then obviously component would have rendered, added as most of the test which I referred starts with checking if the component is rendered

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 2 above)
Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:89]: Have updated the spec with a check for git-type should not be rendered along with above. Does it makes sense now?

**Comment by invincibleJai** [comment]
@christianvogt thanks for the review. Have updated specs and added few more, PTAL

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 1 above)
Comment (christianvogt) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:100]: Asserting that the text should be `Create` is not correct. You're creating a unit test that is directly coupled to the output of the static HTML part of the component and will never fail unless the text is changed.
If we want to start having tests to ensure that static markup or copy don't change, we should take snapshots of the entire component. However we'd end up changing the unit test on every update to the component.

A valid test here would be to locate the `submit` button and assert that it has the `disabled` attribute. You should fix the component which currently applies a `className` of `disabled` which is incorrect behavior right now. Part of writing unit tests is to validate the correctness of the component you are testing.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 2 above)
Comment (christianvogt) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:89]: The only part that is value here as a unit test is that the git type form field is not present. The rest is noise.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx (hunk 1 above)
Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx:100]: ideally className "disabled" should update the attribute "disabled" to true, which is not the case as in below

```
<Button type="submit" bsStyle="primary" className="disabled" active={false} block={false} disabled={false} bsClass="btn">
            Create
 </Button>
```

As suggested will update the component with attribute `disabled`

File: frontend/public/extend/devconsole/components/ImportFlowForm/ImportFlowForm.tsx [hunk 4]
```diff
@@ -586,6 +586,7 @@ export class ImportFlowForm extends React.Component<Props, State> {
             type="submit"
             bsStyle="primary"
             className={this.disableSubmitButton() ? 'disabled' : ''}
+            disabled={this.disableSubmitButton()}
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/ImportFlowForm/ImportFlowForm.tsx:588]: Remove the addition of the `disabled` class name as it's unnecessary for buttons.

Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/ImportFlowForm.tsx:588]: @christianvogt strange but boolean attribute not working as expected https://github.com/facebook/react/pull/993

will put a fix post testing

Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/ImportFlowForm/ImportFlowForm.tsx:588]: @christianvogt false alarm, it's working as expected

**Comment by invincibleJai** [comment]
@christianvogt have updated PR with some tests for pipelines as well PTAL, in case you missed.

File: frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx [hunk 5]
```diff
@@ -78,6 +78,27 @@ describe('ImportFlowForm: ', () => {
...
+  }
+
+  it("shouldn't render git-type", () => {
+    const importFormWrapper = renderImportForm();
+    expect(importFormWrapper.find("FormGroup[controlId='import-git-type']").length).toBe(0);
+  });
+
+  it("AppNameSelector to be present", () => {
+    const importFormWrapper = renderImportForm();
+    expect(importFormWrapper.find("AppNameSelector").length).toBe(1);
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/ImportFlowForm/__tests__/ImportFlowForm.spec.tsx]: For future reference:
With `jest` you can assert array lengths `.toHaveLength(1)`.
https://jestjs.io/docs/en/expect.html#tohavelengthnumber

**Review by christianvogt — APPROVED** [review]

**Pull request merged**